gamearea.py

```diff
@@ -208,25 +215,30 @@ def __motion_cb(self, widget, event):
 
     def __press_cb(self, widget, event):
         self.clicked = [event.x, event.y]
+        try:
+            if self.level_data["type"] == GameType.DIVIDED_SCREEN:
+                if self.over_cat is not None:
+                    self.selected_cat = self.over_cat
+                    self.bring_to_front(self.selected_cat)
 
-        if self.level_data["type"] == GameType.DIVIDED_SCREEN:
-            if self.over_cat is not None:
-                self.selected_cat = self.over_cat
-                self.bring_to_front(self.selected_cat)
+                    self.redraw()
 
-                self.redraw()
+            if self.level_data["type"] == GameType.ROWS:
+                self.selected_option = self.over_option
+        except KeyError:
+            return 0
 
-        if self.level_data["type"] == GameType.ROWS:
-            self.selected_option = self.over_option
 
     def __release_cb(self, widget, event):
         self.clicked = []
         self.selected_cat = None
-
-        if self.level_data["type"] == GameType.ROWS:
-            if self.selected_option is not None:
-                self.count = 0
-                self.redraw()
+        try:
+            if self.level_data["type"] == GameType.ROWS:
+                if self.selected_option is not None:
+                    self.count = 0
+                    self.redraw()
+        except KeyError:
+            return 0
```

Comment:
I think an if-condition check is better than exception. @chimosky any suggestions?

```suggestion
        if "type" not in self.level_data:
            return 0
        if self.level_data["type"] == GameType.ROWS:
            if self.selected_option is not None:
                self.count = 0
                self.redraw()
```

Reply:
Agreed, an if else clause would do in this case and also prevent the `KeyError` from happening, L235 in your change can better be suited as;
```python3
if "type" not in self.level_data.keys():
```
for specificity.